scripts/build_brain.py

```
@@ -0,0 +1,711 @@
+    
+    with open(output_path, "w") as f:
+        json.dump(brain_dump, f, indent=2)
+    
+    print(f"\n✓ Brain dump saved to {output_path}")
+    print(f"  Nodes: {len(nodes)}, Edges: {len(edges)}, Clusters: {len(clusters)}")
+    return brain_dump
+
+
+def build_brain_for_project(
+    project_path: Path,
+    output_path: Path = None,
```

Ruff (0.14.10)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)